mmv1/third_party/terraform/utils/mtls_util.go

```diff
@@ -0,0 +1,49 @@
+package google
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"google.golang.org/api/option/internaloption"
+	"google.golang.org/api/transport"
+)
+
+func isMtls() bool {
+	regularEndpoint := "https://mockservice.googleapis.com/v1/"
+	mtlsEndpoint := getMtlsEndpoint(regularEndpoint)
+	_, endpoint, err := transport.NewHTTPClient(context.Background(),
+		internaloption.WithDefaultEndpoint(regularEndpoint),
+		internaloption.WithDefaultMTLSEndpoint(mtlsEndpoint),
+	)
+	if err != nil {
+		return false
+	}
+	isMtls := strings.Contains(endpoint, "mtls")
+	return isMtls
+}
+
+func getMtlsEndpoint(baseEndpoint string) string {
+	u, err := url.Parse(baseEndpoint)
+	if err != nil {
+		if strings.Contains(baseEndpoint, ".googleapis") {
+			return strings.Replace(baseEndpoint, ".googleapis", ".mtls.googleapis", 1)
+		}
+		return baseEndpoint
+	}
+	portParts := strings.Split(u.Host, ":")
```

Comment:
Do we have any endpoints that include the port in this format? This code is only run on our default base paths, so why would we need to include this?

Reply:
this is a forked version of the function from https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blob/main/dcl/transport.go#L234 .. I can revise it to be more specific to our scenario ?